jme3-effects/src/main/java/com/jme3/post/filters/ContrastAdjustmentFilter.java

```diff
@@ -0,0 +1,309 @@
+    }
+
+    /**
+     * Instantiates a color contrast filter with a default brightness and default scale.
+     * Default values :
+     * - Brightness = 1.0f (normal).
+     * - Scale = 1.0f.
+     *
+     * @param exp_r the red color exponent.
+     * @param exp_b the blue color exponent.
+     * @param exp_g the green color exponent.
+     */
+    public ContrastAdjustmentFilter(float exp_r, float exp_b, float exp_g) {
+        super(ContrastAdjustmentFilter.class.getName());
+        this.exp_r = exp_r;
+        this.exp_g = exp_g;
+        this.exp_b = exp_b;
+    }
+
+    /**
+     * Instantiates a color contrast filter by adjusting different parameters.
+     *
+     * @param exponents  the exponents applied to the colors in order, where x = r, y = g, z = b.
+     * @param brightness the brightness applied to the textures in order, where x = minBrightness, y = maxBrightness.
+     * @param scales     the final pass scales that would be applied on the color channels before being processed in order, where x = scale_r, y = scale_g, z = scale_b.
+     */
+    public ContrastAdjustmentFilter(Vector3f exponents, Vector2f brightness, Vector3f scales) {
+        super(ContrastAdjustmentFilter.class.getName());
+        this.exp_r = exponents.x;
+        this.exp_b = exponents.y;
+        this.exp_g = exponents.z;
+        this.minBrightness = brightness.x;
+        this.maxBrightness = brightness.y;
+        this.scale_r = scales.x;
+        this.scale_b = scales.y;
+        this.scale_g = scales.z;
+    }
+
+    /**
+     * Sets the exponents used to adjust the contrast of the color channels.
+     * Default values are 2.2f.
+     *
+     * @param exp_r the red channel exponent.
+     * @param exp_b the blue channel exponent.
+     * @param exp_g the green channel exponent.
+     */
+    public void setExponents(float exp_r, float exp_b, float exp_g) {
+        this.exp_r = exp_r;
+        this.exp_b = exp_b;
+        this.exp_g = exp_g;
+
+        if (material == null) {
+            return;
+        }
+        //different channels exp for different transfer functions
+        material.setFloat("exp_r", exp_r);
+        material.setFloat("exp_g", exp_g);
+        material.setFloat("exp_b", exp_b);
+    }
+
+    /**
+     * Retrieves the blue channel exponent.
+     * Default value = 2.2.
+     * @return the blue channel exponent.
+     */
+    public float getExp_b() {
+        return exp_b;
+    }
+
+    /**
+     * Retrieves the green channel exponent.
+     * Default value = 2.2.
+     * @return the green channel exponent.
+     */
+    public float getExp_g() {
+        return exp_g;
+    }
+
+    /**
+     * Retrieves the red channel exponent.
+     * Default value = 2.2.
+     * @return the red channel exponent.
+     */
+    public float getExp_r() {
+        return exp_r;
+    }
+
+    /**
+     * Sets the color channels brightness using a maxValue and a minValue based on this equation :
```

Comment:
This is misleading because it doesn't set the color channels' brightness. It sets the range of input levels.